src/demo/models/predict.js

```diff
@@ -61,3 +75,22 @@ const onClickPredict = () => {
   state = setState({trainingIndex: state.trainingIndex});
   asyncSetUiElements(state);
 };
+
+const showCode = show => {
+  setState({showCode: show});
+  if (show) {
+    $('#hide-code').show();
+    $('#show-code').hide();
+    // Fade in the code.  Stop the current and any pending jquery animations.
+    $('#code')
+      .stop(true, true)
+      .fadeIn();
+  } else {
+    $('#show-code').show();
+    $('#hide-code').hide();
+    // Fade out the code.  Stop the current and any pending jquery animations.
+    $('#code')
+      .stop(true, true)
+      .fadeOut();
+  }
+};
```

Comment:
To be clear I do not like writing UI code like this :)  If we have much more of this kind of UI, then we have a great candidate for either React or a React-like system that re-renders UI based on state rather than dynamically adjusting visibility of different elements in an event handler.